server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -372,7 +372,7 @@ private void validateExistingIndex(IndexMetaData currentIndexMetaData, IndexMeta
                     // Make sure that the number of shards is the same. That's the only thing that we cannot change
                     if (currentIndexMetaData.getNumberOfShards() != snapshotIndexMetaData.getNumberOfShards()) {
                         throw new SnapshotRestoreException(snapshot, "cannot restore index [" + renamedIndex + "] with [" + currentIndexMetaData.getNumberOfShards() +
-                                "] shard from snapshot with [" + snapshotIndexMetaData.getNumberOfShards() + "] shards");
+                                "] shard from snapshot with index [" + snapshotIndexMetaData.getIndex().getName() + "] with [" + snapshotIndexMetaData.getNumberOfShards() + "] shards");
```

Comment:
how about `shardS from snapshot OF index [`?

Reply:
@bleskes, that looks better, made the change.